deployment/multiclient.go

```diff
@@ -51,28 +49,26 @@ type MultiClient struct {
 	chainName   string
 }
 
-func NewMultiClient(lggr logger.Logger, rpcs []RPC, opts ...func(client *MultiClient)) (*MultiClient, error) {
-	if len(rpcs) == 0 {
-		return nil, errors.New("No RPCs provided, need at least one")
+func NewMultiClient(lggr logger.Logger, rpcsCfg RPCConfig, opts ...func(client *MultiClient)) (*MultiClient, error) {
+	if len(rpcsCfg.RPCs) == 0 {
+		return nil, errors.New("no RPCs provided, need at least one")
 	}
 	mc := MultiClient{lggr: lggr}
-	clients := make([]*ethclient.Client, 0, len(rpcs))
-	for _, rpc := range rpcs {
-		client, err := ethclient.Dial(rpc.WSURL)
-		if err != nil {
-			return nil, fmt.Errorf("failed to dial ws url '%s': %w", rpc.WSURL, err)
-		}
-		id, err := client.ChainID(context.Background())
+	clients := make([]*ethclient.Client, 0, len(rpcsCfg.RPCs))
+	for i, rpc := range rpcsCfg.RPCs {
+		client, err := mc.dialWithRetry(rpc, lggr)
 		if err != nil {
```

Comment:
I believe we can avoid the call to fetch the ChainID at line 65 by injecting the ChainName directly into the RPCConfig struct. Since the ChainID is already present in the RPC configuration TOML files in CLD, we can simply leverage it to query the chain selectors module there.